packages/SidePanel/src/helpers.js

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+
+export function extractChildren(children, types) {
+  const _children = [];
+  const components = {};
+  if (Array.isArray(types)) {
+    React.Children.toArray(children).forEach(child => {
+      if (child.type && types.includes(child.type.componentType)) {
+        if (Object.prototype.hasOwnProperty.call(components, child.type.componentType)) {
+          const childs = Array.isArray(components[child.type.componentType])
+            ? [child, ...components[child.type.componentType]]
+            : [child, components[child.type.componentType]];
+
+          components[child.type.componentType] = childs;
+        } else {
+          components[child.type.componentType] = child;
```

Comment:
don't think so because I'm checking if the value of components['somevalue'] is an array. In case is not I want to keep just 1 component there instead of converting it into an array.